qiskit/transpiler/passes/optimization/template_matching/template_substitution.py

```diff
@@ -563,18 +563,18 @@ def dummy_parameter():
         for circuit_param, template_param in zip(circuit_params, template_params):
             if isinstance(template_param, ParameterExpression):
                 if isinstance(circuit_param, ParameterExpression):
-                    circ_param_sym = circuit_param.sympify()
+                    circ_param_sym = sym.sympify(circuit_param.sympify())
```

Comment:
Same here, is the additional `sym.sympify` required? It would be good to minimize API changes if possible 🤔